examples/pitr/variables.tf

```diff
@@ -45,6 +45,12 @@ variable "pitr_id" {
   description = "The ID of the source deployment PostgreSQL instance that you want to recover back to. The PostgreSQL instance is expected to be in an up and in running state."
 }
 
+variable "pitr_time" {
+  type        = string
+  description = "The timestamp in UTC format (%Y-%m-%dT%H:%M:%SZ) for any time in the last 7 days that you want to restore to. If no value is passed, earliest_point_in_time_recovery_time will be used as pitr_time. To retrieve the timestamp, run the command (ibmcloud cdb postgresql earliest-pitr-timestamp <deployment name or CRN>). For more info on Point-in-time Recovery, see https://cloud.ibm.com/docs/databases-for-postgresql?topic=databases-for-postgresql-pitr"
+}
```

Comment:
the var `pitr_time` has no default value meaning its a required variable. Is this what you want?

Reply:
Yes, if not value is passed it will pick up the arliest_point_in_time_recovery_time as PITR time. Mentioned the same in the variable description.

Reply:
But this won't let you pass no value - its now a required input. You would need to add a default value of `null` to allow that

Reply:
The point of the feature is to allow the database consumer to create a new database at a point in time. As such, the time is critical and therefor in any reasonable use case would require a variable for the user to set.

With some inference from the documentation, the PITR is a rolling 7 day window and there is poorly defined/odd behaviour with some time selections.
There is a special case, you can query called `earliest_point_in_time_recovery_time`. If you specify before this time, it may or may not work and it may give you earliest.
There is another special case in called `latest_point_in_time_recovery_time`. In the UI this a radio button. In the CLI and API it is passing a single space instead of a time. There is equivalent for this in terraform (that we have found).

Reply:
@Ak-sky I still don't think this statement is correct ?

> If no value is passed, latest_point_in_time_recovery_time will be used as pitr_time.

No value would indicate passing `null`. But am I right in thinking this should actually say empty string?